File: week10_6_11/BOJ_20413_민지(해결중).java
package week10_6_11;

import java.util.HashMap;
import java.util.Scanner;

public class BOJ_20413_민지(해결중) {
  public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		
		HashMap<Character, Integer> tierMax = new HashMap<>();
		tierMax.put('B', sc.nextInt()-1);
		tierMax.put('S', sc.nextInt()-1);
		tierMax.put('G', sc.nextInt()-1);
		tierMax.put('P', sc.nextInt()-1);
		
		String tierAll = sc.next();
		sc.close();
		
		
		int current = tierMax.get(tierAll.charAt(0));
		int total = current;
		
		
		
		for (int i=1; i<n; i++) {
			char tier = tierAll.charAt(i);
			if (tier == 'D') {
				current = 500; 
			} else {
				current = tierMax.get(tier) - current;
			}
			total += current;
		}
		
		System.out.println(total);
		
	}
}

File: week10_6_1/BOJ_20413_민지(해결중).py
N = int(input())
limits = list(map(int, input().split()))
tier = input()

# 각 티어에서 과금할 수 있는 최대 금액을 dict에 저장
tier_max = {'B': limits[0]-1, 'S': limits[1]-1, 'G': limits[2]-1, 'P': limits[3]-1, 'D': 500}


total = current = tier_max[tier[0]]

for i in range(1, len(tier)):
    if tier[i] == 'D':
        current = 500
    elif current == 500:
        total -= 500
        current = tier_max[tier[i]]
    else:
        current = tier_max[tier[i]] - current
    total += current

print(total)